src/document.c

```diff
@@ -825,7 +825,8 @@ GeanyDocument *document_new_file_if_non_open(void)
 }
 
 
-/**
+/** @girskip
```

Comment:
The ownership is one essential difference, yes. I'd rather not have the bindings think they received a reference they own (although the unref/free function does nothing anyway).

The other significant difference is that, depending on the language, the syntax to invoke a constructor is different (not for python, but think of Vala `new XXX.file()` vs `XXX.new_file()`).

Maybe other languages put other constraints on constructors. I really wouldn't want GI think it's a constructor when it isn't (in GI's sense).